File: ASCOM.Utilities/ASCOM.Utilities/AlpacaSupport/WebClient.vb
Imports System
Imports System.Net

Friend Class WebClientWithTimeOut
    Inherits Net.WebClient

    Public Property Timeout As Integer

    Protected Overrides Function GetWebRequest(ByVal uri As Uri) As WebRequest
        Dim webRequest As WebRequest = MyBase.GetWebRequest(uri)
        webRequest.Timeout = Timeout
        CType(webRequest, HttpWebRequest).ReadWriteTimeout = Timeout
        Return webRequest
    End Function
End Class

File: ASCOM.Utilities/ASCOM.Utilities/AlpacaSupport/WebClient.cs
using System;
using System.Net;

namespace ASCOM.Utilities
{

    internal class WebClientWithTimeOut : WebClient
    {

        public int Timeout { get; set; }

        protected override WebRequest GetWebRequest(Uri uri)
        {
            var webRequest = base.GetWebRequest(uri);
            webRequest.Timeout = Timeout;
            ((HttpWebRequest)webRequest).ReadWriteTimeout = Timeout;
            return webRequest;
        }
    }
}